Modules/SearchCityViewController/SearchCityViewModel.swift

```diff
@@ -0,0 +1,142 @@
+//
+//  SearchCityViewModel.swift
+//  WeatherDemoApp
+//
+//  Created by ahmed on 04/04/2024.
+//
+
+import Foundation
+import Combine
+
+final class SearchCityViewModel: BaseViewModel {
+    
+    /// output
+    let localStorageError = PassthroughSubject<LocalStorageError, Never>()
+    let serverError = PassthroughSubject<GenericServerErrorModel, Never>()
+
+    private var cities = [LocalStorageCity]()
+    private var disposables = Set<AnyCancellable>()
+
+    /// callbacks
+    var reloadTableView: VoidCallback!
```

Comment:
```suggestion
    private var onReloadData: VoidCallback!
```
Naming can be view-agnostic. What if you decided to use a collection view instead? You will have to rename all the properties.